### PR TITLE
MQE-1969: Updated CHANGELOG.md and bumped MFTF package version

### PR DESCRIPTION
**Pull request by jilu1**
### Fixed Issues (if relevant)

1. magento/magento2-functional-testing-framework#<issue_number>: Issue title
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,39 @@
 Magento Functional Testing Framework Changelog
 ================================================
+
+2.6.0
+-----
+
+* Traceability
+    * MFTF generated cest files are fully compatible for Codeception `dry-run`.
+* Modularity
+    * `mftf generate:tests` and `mftf run:test` commands now accept suite scoped test names in format `[<suitename:>testname]...`.
```
Comment (KevinBKozan) [inline CHANGELOG.md]: Change the syntax example to be `[suitename:testname]...`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,39 @@
...
+* Maintainability
+    * Support `deprecated` syntax for the following test entities:
+        * Test
+        * Action Group
+        * Data
+        * Page
+        * Section
+        * Section Element
+        * Suite
+
```
Comment (KevinBKozan) [inline CHANGELOG.md]: No empty line, include line 21 in the same bulleted list.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,39 @@
...
+        * Page
+        * Section
+        * Section Element
+        * Suite
+        
+        See DevDocs for details 
+    * Improved `mftf static-checks` command to allow executing all or specific static checks.
+    * Added a new static check that checks and reports unused arguments in action groups.
+* Customizability
+    * AWS Secrets Manager is added as an additional credential storage.
```
Comment (KevinBKozan) [inline CHANGELOG.md]: `... Manager has been added as an...`

File: CHANGELOG.md (hunk 1 above)
Comment (jilu1) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 2 above)
Comment (jilu1) [reply CHANGELOG.md]: Changed

File: CHANGELOG.md (hunk 3 above)
Comment (jilu1) [reply CHANGELOG.md]: Fixed

**Pull request merged**